cranelift/codegen/src/machinst/lower.rs

```diff
@@ -377,12 +377,33 @@ impl<'func, I: VCodeInst> Lower<'func, I> {
             }
         }
 
-        // Make a sret register, if one is needed.
+        // Find the sret register, if it's used.
         let mut sret_reg = None;
         for ret in &vcode.abi().signature().returns.clone() {
             if ret.purpose == ArgumentPurpose::StructReturn {
-                assert!(sret_reg.is_none());
-                sret_reg = Some(vregs.alloc(ret.value_type)?);
+                if let Some(entry_bb) = f.stencil.layout.entry_block() {
```

Comment:
I'm inclined to suggest `unwrap` here. I think if we're trying to lower a function that has no entry block, something has gone very wrong.

Reply:
I agree. The other place we use if-let for the entry block is `gen_arg_setup`, perhaps that one should change as well?

Reply:
I was hoping to find that `gen_arg_setup` is dominated by something that panics if there's no entry block. I got as far as `cranelift/codegen/src/dominator_tree.rs` where `compute_postorder` just silently returns without doing anything if there's no entry block. So in that case it looks like we just generate an empty block lowering order and proceed without complaint.

Since I can't quickly prove that lowering isn't expected to work when there's no entry block, I'm going to appeal to authority and ask @cfallin if there's some reason we should be supporting that case.

Reply:
I *think* that the verifier should verify that there is an entry block? If not, it seems like a reasonable thing to add. I don't think we should expend any unnecessary energy doing special things for empty function bodies; they have no meaning/semantics anyway.